tensorflow/python/ops/check_ops.py

```diff
@@ -1942,6 +1942,12 @@ def _get_diff_for_monotonic_comparison(x):
   is_shorter_than_two = math_ops.less(array_ops.size(x), 2)
   short_result = lambda: ops.convert_to_tensor([], dtype=x.dtype)
 
+  # If x is insigned then cast into signed to avoid overflow.
+  # For example overflow always makes  math.is_non_decreasing to True.
+  # Below code avoids it.
+  if x.dtype in [dtypes.uint8, dtypes.uint16, dtypes.uint32, dtypes.uint64]:
```

Comment:
This will reduce the valid range of uint64, and will lead to unnecessary widening in all other types.  It would be better to fix the issue in the underlying implementation.